sway/tree/workspace.c

```diff
@@ -334,15 +334,19 @@ char *workspace_next_name(const char *output_name) {
 	if (target != NULL) {
 		return target;
 	}
-	// As a fall back, get the current number of active workspaces
-	// and return that + 1 for the next workspace's name
-	int ws_num = root->outputs->length;
-	int l = snprintf(NULL, 0, "%d", ws_num);
-	char *name = malloc(l + 1);
-	if (!sway_assert(name, "Could not allocate workspace name")) {
-		return NULL;
-	}
-	sprintf(name, "%d", ws_num++);
+	// As a fall back, use the next available number
+	char *name = NULL;
+	int ws_num = 1;
+	do {
+		if (!name || ws_num % 10 == 0) {
+			int l = snprintf(NULL, 0, "%d", ws_num);
+			name = realloc(name, l + 1);
```

Comment:
Instead, we could just have a 12-byte array on the stack.

Reply:
Changed